examples/api/lib/widgets/animated_size/animated_size.0.dart

```diff
@@ -11,47 +11,62 @@ void main() => runApp(const AnimatedSizeExampleApp());
 class AnimatedSizeExampleApp extends StatelessWidget {
   const AnimatedSizeExampleApp({super.key});
 
+  static const Duration duration = Duration(seconds: 1);
+  static const Curve curve = Curves.easeIn;
+
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
       home: Scaffold(
         appBar: AppBar(title: const Text('AnimatedSize Sample')),
         body: const Center(
-          child: AnimatedSizeExample(),
+          child: AnimatedSizeExample(
+            duration: duration,
+            curve: curve,
+          ),
         ),
       ),
     );
   }
 }
 
 class AnimatedSizeExample extends StatefulWidget {
-  const AnimatedSizeExample({super.key});
+  const AnimatedSizeExample({
+    required this.duration,
+    required this.curve,
+    super.key,
+  });
+
+  final Duration duration;
+
+  final Curve curve;
 
   @override
   State<AnimatedSizeExample> createState() => _AnimatedSizeExampleState();
 }
 
 class _AnimatedSizeExampleState extends State<AnimatedSizeExample> {
-  double _size = 50.0;
-  bool _large = false;
-
-  void _updateSize() {
-    setState(() {
-      _size = _large ? 250.0 : 100.0;
-      _large = !_large;
-    });
-  }
+  bool _isSelected = false;
 
   @override
   Widget build(BuildContext context) {
     return GestureDetector(
-      onTap: () => _updateSize(),
+      onTap: () {
+        setState(() {
+          _isSelected = !_isSelected;
+        });
+      },
       child: ColoredBox(
         color: Colors.amberAccent,
         child: AnimatedSize(
-          curve: Curves.easeIn,
-          duration: const Duration(seconds: 1),
-          child: FlutterLogo(size: _size),
+          duration: widget.duration,
+          curve: widget.curve,
+          child: SizedBox.square(
+            dimension: _isSelected ? 250.0 : 100.0,
+            child: const Center(
+              child: FlutterLogo(size: 75.0),
+            ),
+          ),
```

Comment:
Any reason to not animate the `FlutterLogo` the way it was done before? I think the visual effect made sense and the sample will be a little easier to understand when removing the `SizedBox`.

Reply:
@bleroux 
This was done because `FlutterLogo` already has size animation under the hood, and the main goal of the example is to show `AnimatedSize` usage. You can try the following code sample to verify that `FlutterLogo` animates without the usage of `AnimatedSize`.

<details closed><summary>Code sample</summary>

```dart
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter/material.dart';

/// Flutter code sample for [AnimatedSize].

void main() => runApp(const AnimatedSizeExampleApp());

class AnimatedSizeExampleApp extends StatelessWidget {
  const AnimatedSizeExampleApp({super.key});

  static const Duration duration = Duration(seconds: 1);
  static const Curve curve = Curves.easeIn;

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: const Text('AnimatedSize Sample')),
        body: const Center(
          child: AnimatedSizeExample(
            duration: duration,
            curve: curve,
          ),
        ),
      ),
    );
  }
}

class AnimatedSizeExample extends StatefulWidget {
  const AnimatedSizeExample({
    required this.duration,
    required this.curve,
    super.key,
  });

  final Duration duration;

  final Curve curve;

  @override
  State<AnimatedSizeExample> createState() => _AnimatedSizeExampleState();
}

class _AnimatedSizeExampleState extends State<AnimatedSizeExample> {
  bool _isSelected = false;

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: () {
        setState(() {
          _isSelected = !_isSelected;
        });
      },
      child: Center(
        child: ColoredBox(
          color: Colors.amberAccent,
          child: FlutterLogo(
            size: _isSelected ? 250.0 : 100.0,
          ),
        ),
      ),
    );
  }
}

```

</details>